core/src/main/java/io/undertow/protocols/ajp/AjpResponseParser.java

```diff
@@ -224,11 +224,6 @@ public int getReadBodyChunkSize() {
      */
     public int stringLength = -1;
 
-    /**
-     * The current string being read
-     */
-    public StringBuilder currentString;
```

Comment:
You can't remove currentString here. The parser is invoked multiple times as bytes are made available and, when the last byte is read, it wil need the whole currentString to form a parsed result, it will neeed information from previous parse invocations. Otherwise, you risk losing bytes. It all depends on how many bytes the parser has received on every invocation.